app/Console/Commands/KeyGenerateCommand.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace App\Console\Commands;
+
+use Illuminate\Encryption\Encrypter;
+use Safe\Exceptions\UrlException;
+
+class KeyGenerateCommand extends \Illuminate\Foundation\Console\KeyGenerateCommand
+{
+	/**
+	 * Set the application key in the environment file.
+	 *
+	 * @param string $key
+	 *
+	 * @return bool
+	 *
+	 * @throws UrlException
+	 */
+	protected function setKeyInEnvironmentFile($key): bool
+	{
+		$currentKey = $this->laravel['config']['app.key'];
+		if (str_starts_with($currentKey, 'base64:')) {
+			$currentKey = substr($currentKey, 7);
+		}
+		$supported = Encrypter::supported(\Safe\base64_decode($currentKey), $this->laravel['config']['app.cipher']);
```

Comment:
Why is `\base64_decode` outside of the condition which checks whether the value starts with `base64:`? I haven't looked up where the key is used inside the framework, so it might be correct and the key is always encoded in Base64 even if it does not start with `base64`. But is looks suspicious, I would have expected `base64_encode` inside the condition.